packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php

```diff
@@ -333,7 +333,7 @@ private function processUnreachableStatementNode(
         }
 
         $stmtKey = $unreachableStatementNode->getAttribute(AttributeKey::STMT_KEY);
-        $totalKeys = $parentNode->stmts === null ? 0 : count($parentNode->stmts);
+        $totalKeys = $parentNode->stmts === null ? 0 : count((array) $parentNode->stmts);
```

Comment:
I guess you cast here because `$parentNode->stmts` might be `null`.

I think you should early return instead when its `null`, because the loop below also doesn't make sense when `$parentNode->stmts=null`

Reply:
Return early is not make original bug disappear.

The stmts only 2 options: array or null. The cast should not needed, even with ternary else.

Reply:
On abstract method, stmts is null

Reply:
Yes, but it is checked in first part of ternary already: 

```php
***$parentNode->stmts === null*** ? 0 : count($parentNode->stmts);
```

Reply:
Exactly :+1:  

But the PHPStan sees it as an `array|null` even after the ternary check. It should see only `array`